packages/pigeon/lib/dart_generator.dart

```diff
@@ -8,13 +8,45 @@ import 'generator_tools.dart';
 /// Options that control how Dart code will be generated.
 class DartOptions {
   /// Constructor for DartOptions.
-  DartOptions({this.isNullSafe = true, this.copyrightHeader});
+  const DartOptions({this.isNullSafe = true, this.copyrightHeader});
 
   /// Determines if the generated code has null safety annotations (Dart >=2.12 required).
-  bool isNullSafe;
+  final bool isNullSafe;
 
   /// A copyright header that will get prepended to generated code.
-  Iterable<String>? copyrightHeader;
+  final Iterable<String>? copyrightHeader;
+
+  /// Creates a [DartOptions] from a Map representation where:
+  /// `x = DartOptions.fromMap(x.toMap())`.
+  static DartOptions fromMap(Map<String, Object> map) {
+    return DartOptions(
+      isNullSafe:
+          // ignore: avoid_bool_literals_in_conditional_expressions
+          map.containsKey('isNullSafe') ? (map['isNullSafe'] as bool?)! : true,
+      copyrightHeader: map.containsKey('copyrightHeader')
+          ? map['copyrightHeader'] as Iterable<String>?
+          : null,
+    );
+  }
+
+  /// Converts a [DartOptions] to a Map representation where:
+  /// `x = DartOptions.fromMap(x.toMap())`.
+  Map<String, Object> toMap() {
+    final Map<String, Object> result = <String, Object>{};
+    if (isNullSafe != null) {
+      result['isNullSafe'] = isNullSafe;
+    }
+    if (copyrightHeader != null) {
+      result['copyrightHeader'] = copyrightHeader!;
+    }
+    return result;
+  }
+
+  /// Overrides any non-null parameters from [options] into this to make a new
```

Comment:
I'm having a hard time parsing "overrides ... from ... into". What is the direct object of "overrides"?

Would "Returns a new [DartOptions] based on this object, but with any non-null values in [options] replacing the originals" capture the intent?

Reply:
updated